packages/trpc/server/routers/viewer/bookings/get.handler.ts

```diff
@@ -384,41 +383,62 @@ export async function getBookings({
 
   // Now enrich bookings with relation data. We could have queried the relation data along with the bookings, but that would cause unnecessary queries to the database.
   // Because Prisma is also going to query the select relation data sequentially, we are fine querying it separately here as it would be just 1 query instead of 4
-  const bookings = (
-    await prisma.booking.findMany({
-      where: {
-        id: {
-          in: plainBookings.map((booking) => booking.id),
+  const teamAdminOrOwnerCache = new Map<number, boolean>();
+
+  const bookings = await Promise.all(
+    (
+      await prisma.booking.findMany({
+        where: {
+          id: {
+            in: plainBookings.map((booking) => booking.id),
+          },
         },
-      },
-      select: bookingSelect,
-      // We need to get the sorted bookings here as well because plainBookings array is not correctly sorted
-      orderBy,
-    })
-  ).map((booking) => {
-    // If seats are enabled and the event is not set to show attendees, filter out attendees that are not the current user
-    if (booking.seatsReferences.length && !booking.eventType?.seatsShowAttendees) {
-      booking.attendees = booking.attendees.filter((attendee) => attendee.email === user.email);
-    }
+        select: bookingSelect,
+        // We need to get the sorted bookings here as well because plainBookings array is not correctly sorted
+        orderBy,
+      })
+    ).map(async (booking) => {
+      // If seats are enabled and the event is not set to show attendees, filter out attendees that are not the current user
+      if (booking.seatsReferences.length && !booking.eventType?.seatsShowAttendees) {
+        booking.attendees = booking.attendees.filter((attendee) => attendee.email === user.email);
+      }
 
-    const membership = booking.eventType?.team?.members.find((membership) => membership.userId === user.id);
-    const isUserTeamAdminOrOwner =
-      membership?.role === MembershipRole.OWNER || membership?.role === MembershipRole.ADMIN;
+      let isUserTeamAdminOrOwner = false;
+      if (booking.eventType?.team?.id) {
+        const teamId = booking.eventType.team.id;
+        if (teamAdminOrOwnerCache.has(teamId)) {
+          isUserTeamAdminOrOwner = teamAdminOrOwnerCache.get(teamId)!;
+        } else {
+          isUserTeamAdminOrOwner = await prisma.membership
```

Comment:
What's the likelihood of this query running lots of times? I guess it would require a user to have bookings across lots of different team event types?

Reply:
yeah for enterprise plan user or big teams plan users. they are part of many teams and some teams have 200+ members.

Reply:
> I guess it would require a user to have bookings across lots of different team event types?

Yeah. I added `teamAdminOrOwnerCache` to store this membership result.